pkg/database/databaseLayer.go

```diff
@@ -92,7 +92,7 @@ func (db DBClient) UpdateAgentStatus(agent models.Agent) error {
 
 func (db DBClient) GetAllAgents() ([]models.Agent, error) {
 	result := []models.Agent{}
-	err := db.Db.Find(&result)
+	err := db.Db.Model(&models.Agent{}).Where("is_active=?", true).Find(&result)
```

Comment:
This function name suggests that it will return all the agents, but you are querying just for the active agents. Either update the function name to GetActiveAgents or pass the is_active param to GetAllAgents and use it in the query building conditionally.